modules/AlphaWalletFoundation/AlphaWalletFoundation/Types/TokenUriMapSupportable.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  TokenUriMapSupportable.swift
+//  AlphaWalletFoundation
+//
+//  Created by Vladyslav Shepitko on 16.03.2023.
+//
+
+import Foundation
+import BigInt
+
+protocol TokenUriMapSupportable {
+    func map(uri: URL) -> URL?
+}
+
+struct TokenUriMapper: TokenUriMapSupportable {
+    let hostMeppers: [TokenUriMapSupportable]
+
+    func map(uri: URL) -> URL? {
+        hostMeppers.compactMap { $0.map(uri: uri) }.first
```

Comment:
It seems to work, but the logic here looks unintuitive and maybe slower? Can we rewrite it so it's something like this instead?:

```
hostMeppers.first { $0.match(uri: uri) }?.mappedUri
```

It finds the first match, rather than operate on all of them and gets the first

(and typo: `hostMeppers` -> `hostMappers` or `mappings`?)

Reply:
And add a test case?